AnyType.hpp

```diff
@@ -0,0 +1,193 @@
+#include <string>
+#include <typeinfo>
+
+class AnyType {
+private:
+	std::string type;
+	union {
+		bool DataBool;
+		char DataChar;
+		signed char DataSChar;
+		unsigned char DataUnsChar;
+		wchar_t DataWCharT;
+		char16_t DataChar16T;
+		char32_t DataChar32T;
+		int DataInt;
+		unsigned int DataUnsInt;
+		short int DataShortInt;
+		unsigned short int DataUnsShortInt;
+		long int DataLInt;
+		unsigned long int DataUnsLInt;
+		long long int DataLLInt;
+		unsigned long long int DataUnsLLInt;
+		float DataFloat;
+		double DataDouble;
+		long double DataLDouble;
+	};
+public:
+	AnyType();
+	void Swap(AnyType&);
+	std::string GetType();
+	
+	template <typename T>
+	AnyType(T data) {

```

Comment:
You can use map instead of this long list of "if" operators. It will provide more flexibility and code will looks better.